qiskit/scheduler/basic_scheduler.py

```diff
@@ -0,0 +1,160 @@
+    """
+    circuit_copy = deepcopy(circuit)
+    circuit_copy.data.reverse()
+    reversed_schedule = greedy_schedule(circuit_copy, backend)
+    total_time = max(inst[0] + inst[1].duration for inst in reversed_schedule.instructions)
+    insts = list(reversed_schedule.instructions)
+    insts.reverse()
+    sched = Schedule()
+    for time, inst in insts:
+        sched |= inst << (total_time - time - inst.duration)
+    return sched
+
+
+def greedy_schedule(circuit: QuantumCircuit,
+                    backend: Optional[BaseBackend] = None,
+                    cmd_def: Optional[CmdDef] = None,
+                    meas_map: Optional[List[List[int]]] = None) -> Schedule:
+    """
+    Return the input circuit's equivalent pulse Schedule by performing the most basic scheduling
+    pass on it. The pulses are "backward scheduled", meaning that pulses are played as early as
+    possible.
+
+    Args:
+        circuit: The quantum circuit to translate
+        backend: A backend instance, which contains hardware specific data required for scheduling
+        cmd_def: Command definition list
+        meas_map: Groups of qubits that get measured together
+    Returns:
+        New forward scheduled Schedule
+    """
+    sched = Schedule(name=circuit.name)
+
+    if cmd_def == None:
+        if backend == None:
+            raise QiskitError("Must supply either a backend or CmdDef for scheduling passes.")
+        defaults = backend.defaults()
+        cmd_def = CmdDef.from_defaults(defaults.cmd_def, defaults.pulse_library)
+    if meas_map == None:
+        if backend == None:
+            raise QiskitError("Must supply either a backend or a meas_map for scheduling passes.")
+        meas_map = format_meas_map(backend.configuration().meas_map)
+
+    qubit_time_available = defaultdict(int)
+    measured_qubits = set()
+
+    def update_times(qubits: List[int], time: int) -> None:
+        """Update the time tracker for all qubits to the given time."""
+        for q in qubits:
+            qubit_time_available[q] = time
+
+    def add_measures():
+        """Based off the Set measured_qubits, add measures to the schedule."""
+        nonlocal sched
+        measures = set()
+        for q in measured_qubits:
+            measures.add(tuple(meas_map[q]))
+        for qubits in measures:
+            time = max(qubit_time_available[q] for q in qubits)
+            cmd = cmd_def.get('measure', qubits)
+            sched |= cmd << time
+            update_times(qubits, time + cmd.duration)
+        measured_qubits.clear()
+
+    for inst, qregs, _ in circuit:
+        qubits = [chan.index for chan in qregs]
+        if isinstance(inst, Barrier):
+            for q in qubits:
+                qubit_time_available[q] = max(qubit_time_available[q] for q in qubits)
+            continue
+        if any(q in measured_qubits for q in qubits):
+            add_measures()
+        if isinstance(inst, Measure):
+            measured_qubits.update(qubits)
```

Comment:
Why handle measurements separately, rather than in the same way as other instructions?
You can actually modify the loop to:
```     for inst, qubits, clbits in circuit:```
and you would have all the qubits and clbits that any instruction acts on. Here it seems like the clbits of the measurements are lost.

Reply:
This is tricky because clbits are handled by providing additional metadata in the qobj header. A clbit does not necessarily correspond to a memory_slot. One approach we have discussed is optionally returning the qobj header metadata for the handling of the ClassicalRegister to later be included at assembly time.

Reply:
I still don't really know what to do with the classical registers, it's true.

As for the measurement part, what would you suggest for something like:
```
qc.measure(q[0], c[0])
qc.measure(q[1], c[1])
```
? I thought we would want to translate this to a single Acquire(q0, q1) pulse. If we treated this like any inst, then for a measure map like `[[0, 1]]` I was getting Acquire(q0, q1) followed by another Acquire(q0, q1), which I didn't think was the desired behavior.

Reply:
For now, we could translate as is with multiple acquires. It would be up to the backend to support this.

Reply:
Can you guys re-review this section carefully? I merged lists of measures within a measure group. I think it more accurately reflects what the circuit writer was hoping for. If you don't agree, just lmk and we can change it :)